### PR TITLE
Moved handle Click logic to impl, added link support in topic detail screen

### PR DESCRIPTION
**Pull request by varunasingh**

### CODE REVIEW

File: app-android/src/main/java/com/ustadmobile/port/android/view/DiscussionPostDescriptionRecyclerAdapter.kt [hunk 1]
```diff
@@ -1,45 +1,98 @@
...
+                this.setOnLinkClickListener(onClickListener)
+                this.setOnLinkLongClickListener(onLongClickListener)
+
+            }
+
+        Linkify.addLinks(textView, Linkify.ALL)
+
+    }
+
+    val onClickListener: BetterLinkMovementMethod.OnLinkClickListener =
```
Comment (mikedawson) [inline app-android/src/main/java/com/ustadmobile/port/android/view/DiscussionPostDescriptionRecyclerAdapter.kt]: Why not have a class that does this? e.g.

```
class BetterLinkMovementLinkClickListener(val systemImpl: UstadMobileSystemImpl)
```

Comment (mikedawson) [reply app-android/src/main/java/com/ustadmobile/port/android/view/DiscussionPostDescriptionRecyclerAdapter.kt]: e.g. class could then be used by all components that need to listen (instead of repeating the same anonymous listeners)?

File: core/src/androidMain/kotlin/com/ustadmobile/core/impl/UstadMobileSystemImpl.kt [hunk 2]
```diff
@@ -485,6 +485,20 @@ actual open class UstadMobileSystemImpl : UstadMobileSystemCommon() {
         (context as Context).startActivity(intent)
     }
 
+    actual fun handleClickLink(url: String, context: Any){
```
Comment (mikedawson) [inline core/src/androidMain/kotlin/com/ustadmobile/core/impl/UstadMobileSystemImpl.kt]: This code should be in common. The logic will be the same.

Needs jvm tests.

Needs to handle edge case: when link is an ustad link, but a different endpoint

File: app-android/src/main/java/com/ustadmobile/port/android/view/BetterLinkMovementClickListener.kt [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        }
+
+    private val onLongClickListener: BetterLinkMovementMethod.OnLinkLongClickListener =
+        BetterLinkMovementMethod.OnLinkLongClickListener{
+                _,_ ->
+            true
+        }
+
+
+    fun addMovement(textView: TextView?){
```
Comment (mikedawson) [inline app-android/src/main/java/com/ustadmobile/port/android/view/BetterLinkMovementClickListener.kt:30]: Why should the parameter be nullable?

File: app-android/src/main/java/com/ustadmobile/port/android/view/MessagesRecyclerAdapter.kt [hunk 4]
```diff
@@ -47,7 +52,9 @@ class MessagesRecyclerAdapter(
         holder.binding.message = message
 
         holder.binding.itemCommentsListLine2Text.text = message?.messageText
-        addMovement(holder.binding.itemCommentsListLine2Text)
+        val listener = BetterLinkMovementLinkClickListener(systemImpl, accountManager, context)
+        listener.addMovement(holder.binding.itemCommentsListLine2Text)
+        //addMovement(holder.binding.itemCommentsListLine2Text)
```
Comment (mikedawson) [inline app-android/src/main/java/com/ustadmobile/port/android/view/MessagesRecyclerAdapter.kt]: Commented out line

**Review by mikedawson — APPROVED** [review]
